optics-core/src/Optics/Setter.hs

```diff
@@ -153,5 +155,29 @@ mapped :: Functor f => Setter (f a) (f b) a b
 mapped = Optic mapped__
 {-# INLINE mapped #-}
 
+-- | Rewrite by applying a rule everywhere you can. Ensures that the rule cannot
+-- be applied anywhere in the result:
+--
+-- @
+-- propRewriteOf l r x = 'all' ('Data.Just.isNothing' '.' r) ('universeOf' l ('rewriteOf' l r x))
+-- @
+--
+-- Usually 'transformOf' is more appropriate, but 'rewriteOf' can give better
+-- compositionality. Given two single transformations @f@ and @g@, you can
+-- construct @\\a -> f a '<|>' g a@ which performs both rewrites until a fixed
+-- point.
+rewriteOf :: Is k A_Setter => Optic k is a b a b -> (b -> Maybe a) -> a -> b
+rewriteOf l f = go
+  where
+    go = transformOf l (\x -> maybe x go (f x))
+{-# INLINE rewriteOf #-}
+
+-- | Transform every element by recursively applying a given 'Setter' in a
+-- bottom-up manner.
+transformOf :: Is k A_Setter => Optic k is a b a b -> (b -> b) -> a -> b
```

Comment:
Is it deliberate  that the `*OnOf` variants are left out? They are useful when you e.g. want  to transform all `Pat`s inside `Exp`:

```haskell
transformOnOf patExps expPlated modifyPat :: Exp -> Exp where
  patExps :: Setter' Pat Exp
  expPlated :: Setter' Pat Pat
  modifyPlated :: Pat -> Pat
```

Of course I can do `over patExps (transformOf expPlated modifyPlated)`...

Reply:
> Of course I can do over patExps (transformOf expPlated modifyPlated)...

Yeah, that's what was also mentioned in https://github.com/well-typed/optics/pull/379#discussion_r654792249, I think these are a bit excessive.

We can always add them later if there is enough demand.